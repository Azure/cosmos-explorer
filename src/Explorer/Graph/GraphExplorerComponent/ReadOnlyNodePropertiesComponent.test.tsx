import { shallow } from "enzyme";
import React from "react";
import { GraphHighlightedNodeData } from "./GraphExplorer";
import {
  ReadOnlyNodePropertiesComponent,
  ReadOnlyNodePropertiesComponentProps,
} from "./ReadOnlyNodePropertiesComponent";

describe("<ReadOnlyNodePropertiesComponent />", () => {
  const id = "myId";
  const label = "myLabel";
  const mockNode: GraphHighlightedNodeData = {
    id: id,
    label: label,
    properties: {
      key1: ["value1"],
      key2: ["value2"],
    },
    areNeighborsUnknown: false,
    sources: [],
    targets: [],
  };

  it("renders id", () => {
    const props: ReadOnlyNodePropertiesComponentProps = { node: mockNode };
    const wrapper = shallow(<ReadOnlyNodePropertiesComponent {...props} />);
    expect(wrapper.find(".vertexId").text()).toBe(id);
  });

  it("renders label", () => {
    const props: ReadOnlyNodePropertiesComponentProps = { node: mockNode };
    const wrapper = shallow(<ReadOnlyNodePropertiesComponent {...props} />);
    expect(wrapper.find(".vertexLabel").text()).toBe(label);
  });

  it("renders properties (single value)", () => {
    const props: ReadOnlyNodePropertiesComponentProps = { node: mockNode };
    const wrapper = shallow(<ReadOnlyNodePropertiesComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders properties (with multiple values)", () => {
    const mockNode2 = {
      ...mockNode,
      properties: {
        key3: ["abcd", 1234, true, false, undefined, null],
      },
    };
    const props: ReadOnlyNodePropertiesComponentProps = { node: mockNode2 };
    const wrapper = shallow(<ReadOnlyNodePropertiesComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders unicode", () => {
    const mockNode2 = {
      ...mockNode,
      properties: {
        key4: ["あきら, アキラ,安喜良"],
        key5: ["Véronique"],
      },
    };
    const props: ReadOnlyNodePropertiesComponentProps = { node: mockNode2 };
    const wrapper = shallow(<ReadOnlyNodePropertiesComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
