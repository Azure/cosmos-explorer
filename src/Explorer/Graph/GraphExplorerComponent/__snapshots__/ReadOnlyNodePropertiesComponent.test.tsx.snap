// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReadOnlyNodePropertiesComponent /> renders properties (single value) 1`] = `
<table
  className="roPropertyTable propertyTable"
>
  <tbody>
    <tr>
      <td
        className="labelCol"
      >
        id
      </td>
      <td>
        <span
          className="vertexId"
        >
          myId
        </span>
      </td>
    </tr>
    <tr>
      <td
        className="labelCol"
      >
        label
      </td>
      <td>
        <span
          className="vertexLabel"
        >
          myLabel
        </span>
      </td>
    </tr>
    <tr
      key="key1"
    >
      <td
        className="labelCol propertyId"
        title="key1"
      >
        key1
      </td>
      <td
        className="valueCol"
        title="value1"
      >
        <div
          className="propertyValue"
          key="value1"
        >
          value1
        </div>
      </td>
    </tr>
    <tr
      key="key2"
    >
      <td
        className="labelCol propertyId"
        title="key2"
      >
        key2
      </td>
      <td
        className="valueCol"
        title="value2"
      >
        <div
          className="propertyValue"
          key="value2"
        >
          value2
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`<ReadOnlyNodePropertiesComponent /> renders properties (with multiple values) 1`] = `
<table
  className="roPropertyTable propertyTable"
>
  <tbody>
    <tr>
      <td
        className="labelCol"
      >
        id
      </td>
      <td>
        <span
          className="vertexId"
        >
          myId
        </span>
      </td>
    </tr>
    <tr>
      <td
        className="labelCol"
      >
        label
      </td>
      <td>
        <span
          className="vertexLabel"
        >
          myLabel
        </span>
      </td>
    </tr>
    <tr
      key="key3"
    >
      <td
        className="labelCol propertyId"
        title="key3"
      >
        key3
      </td>
      <td
        className="valueCol"
        title="abcd, 1234, true, false, undefined, null"
      >
        <div
          className="propertyValue"
          key="abcd"
        >
          abcd
        </div>
        <div
          className="propertyValue"
          key="1234"
        >
          1234
        </div>
        <div
          className="propertyValue"
          key="true"
        >
          true
        </div>
        <div
          className="propertyValue"
          key="false"
        >
          false
        </div>
        <div
          className="propertyValue"
          key="undefined"
        >
          undefined
        </div>
        <div
          className="propertyValue isNull"
          key="null"
        >
          null
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`<ReadOnlyNodePropertiesComponent /> renders unicode 1`] = `
<table
  className="roPropertyTable propertyTable"
>
  <tbody>
    <tr>
      <td
        className="labelCol"
      >
        id
      </td>
      <td>
        <span
          className="vertexId"
        >
          myId
        </span>
      </td>
    </tr>
    <tr>
      <td
        className="labelCol"
      >
        label
      </td>
      <td>
        <span
          className="vertexLabel"
        >
          myLabel
        </span>
      </td>
    </tr>
    <tr
      key="key4"
    >
      <td
        className="labelCol propertyId"
        title="key4"
      >
        key4
      </td>
      <td
        className="valueCol"
        title="あきら, アキラ,安喜良"
      >
        <div
          className="propertyValue"
          key="あきら, アキラ,安喜良"
        >
          あきら, アキラ,安喜良
        </div>
      </td>
    </tr>
    <tr
      key="key5"
    >
      <td
        className="labelCol propertyId"
        title="key5"
      >
        key5
      </td>
      <td
        className="valueCol"
        title="Véronique"
      >
        <div
          className="propertyValue"
          key="Véronique"
        >
          Véronique
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;
